templates/Collection.h.template

```diff
@@ -74,6 +74,9 @@ public:
   /// number of elements in the collection
   int size() const override final ;
 
+  /// fully qualified type name of elements - with namespace
+  std::string getValueTypeName() const override { return std::string("${classname}") ; } ;
```

Comment:
`${classname}` could also be a made a static member, but since I don't expect many instances of a collection, it is more straightforward like this, I think.